tests/recruitment/candidates/searchCandidateSuite/searchCandidate.spec.ts

```diff
@@ -0,0 +1,328 @@
+
+  test("Filter candidates by Status", async ({ page }) => {
+    const fullName = fullNameCombiner(page, ValidUser);
+    await recruitmentPage.status.click();
+    await recruitmentPage.statusName.click();
+    await recruitmentPage.submitButton.click();
+    await expect(fullName).toBeVisible();
+    const firstName = ValidUser.firstName;
+    const candidateName = fullCandidateName(page, ValidUser);
+    await deleteRecord(
+      fullName,
+      firstName,
+      candidateName,
+      page,
+      recruitmentPage
+    );
+  });
+  test("Filter candidates by Candidate Name", async ({ page }) => {
+    const fullName = fullNameCombiner(page, ValidUser);
+    const firstName = ValidUser.firstName;
+    const candidateName = fullCandidateName(page, ValidUser);
+    await recruitmentPage.candidateField.click();
+    await recruitmentPage.candidateField.fill(firstName); //fill first name
+    await candidateName.click();
+    await recruitmentPage.submitButton.click();
+    await expect(fullName).toBeVisible();
+    await deleteRecord(
+      fullName,
+      firstName,
+      candidateName,
+      page,
+      recruitmentPage
+    );
+  });
+
+  test("Filter candidates by Keywords", async ({ page }) => {
+    const fullName = fullNameCombiner(page, ValidUser);
+    await recruitmentPage.keywords.click();
+    const keyWords = ValidUser.keywords;
+    await recruitmentPage.keywords.fill(keyWords);
+    await recruitmentPage.submitButton.click();
+    await page.waitForTimeout(5000);
+    await expect(fullName).toBeVisible();
+    const firstName = ValidUser.firstName;
+    const candidateName = fullCandidateName(page, ValidUser);
+    await deleteRecord(
+      fullName,
+      firstName,
+      candidateName,
+      page,
+      recruitmentPage
+    );
+  });
+
+  test("Filter candidates by From Date", async ({ page }) => {
+    const fullName = fullNameCombiner(page, ValidUser);
+    const formattedFrom = ValidUser.fromDate.split("T")[0];
+    const formattedDate = ValidUser.dateOfApp.split("T")[0];
+    const formattedTo = ValidUser.toDate.split("T")[0];
+    await recruitmentPage.from.click();
+    await recruitmentPage.from.fill(formattedFrom);
+    await recruitmentPage.submitButton.click();
+    await expect(fullName).toBeVisible();
+    await page.waitForTimeout(5000);
+    const firstName = ValidUser.firstName;
+    const candidateName = fullCandidateName(page, ValidUser);
+    await deleteRecord(
+      fullName,
+      firstName,
+      candidateName,
+      page,
+      recruitmentPage
+    );
+  });
+
+  test("Filter candidates by To Date", async ({ page }) => {
+    const fullName = fullNameCombiner(page, ValidUser);
+    const formattedTo = ValidUser.toDate.split("T")[0];
+    await recruitmentPage.to.click();
+    await recruitmentPage.to.fill(formattedTo);
+    await recruitmentPage.to.click();
+    await recruitmentPage.submitButton.click();
+    await expect(fullName).toBeVisible();
+    await page.waitForTimeout(5000);
```

Comment:
chỗ này hạn chế wait hardcode time như này nha, nên xài wait for element to bi visible r để timeout vào đấy

check fix các chỗ tương tự nha